### PR TITLE
Add support for os_subnet role in linchpin topologies

### LINKED ISSUE #820: Openstack: Networking: Add support for os_subnet module ansible
Add support for os_subnet module ansible in linchpin topologies
ref: https://docs.ansible.com/ansible/2.6/modules/os_subnet_module.html#os-subnet-module
as a part of #786

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary:
- Updated examples and documentation for os_subnet role
- Added test for os_subnet
- Updated playbooks to support os_subnet ansible module
- Updated schema to support os_subnet module

fixes #820

### CODE REVIEW

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
[test]

File: docs/source/openstack.rst [hunk 1]
```diff
@@ -96,6 +96,14 @@ Openstack routers can be provisioned using this resource.
 * :docs1.5:`Topology Example <workspace/topologies/os-router.yml>`
 * `Ansible os_router module <https://docs.ansible.com/ansible/latest/modules/os_router_module.html>`_
 
+os_subnet
+---------
+
+Openstack subnets can be provisioned using this resource.
+* :docs1.5:`Topology Example <workspace/topologies/os-subnet.yml>`
```
Comment (14rcole) [inline docs/source/openstack.rst]: At some point we should come up with an alternative way to do this so that we no longer need the workspace/ directory

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**